projects/ad469x_iio/app/app_config.c

```diff
@@ -0,0 +1,353 @@
+#include "no_os_pwm.h"
+#include "no_os_uart.h"
+#include "common.h"
+
+/******************************************************************************/
+/************************ Macros/Constants ************************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+/* UART init parameters for IIO comm port */
+struct no_os_uart_init_param uart_iio_comm_init_params = {
+	.device_id = UART_MODULE,
+	.asynchronous_rx = false,
+	.baud_rate = IIO_UART_BAUD_RATE,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.irq_id = UART_IRQ,
+#if defined(USE_VIRTUAL_COM_PORT)
+	.platform_ops = &vcom_ops,
+	.extra = &vcom_extra_init_params
+#else
+	.platform_ops = &uart_ops,
+	.extra = &uart_extra_init_params
+#endif
+};
+
+/* UART init parameters for console comm port */
+struct no_os_uart_init_param uart_console_stdio_init_params = {
+	.device_id = UART_MODULE,
+	.asynchronous_rx = false,
+	.baud_rate = IIO_UART_BAUD_RATE,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+#if defined(USE_VIRTUAL_COM_PORT)
+	/* If virtual com port is primary IIO comm port, use physical port for stdio
+	 * console. Applications which does not support VCOM, should not satisfy this
+	 * condition */
+	.platform_ops = &uart_ops,
+	.extra = &uart_extra_init_params
+#else
+#if defined(CONSOLE_STDIO_PORT_AVAILABLE)
+	/* Applications which uses phy COM port as primary IIO comm port,
+	 * can use VCOM as console stdio port provided it is available.
+	 * Else, alternative phy com port can be used for console stdio ops if available */
+	.platform_ops = &vcom_ops,
+	.extra = &vcom_extra_init_params
+#endif
+#endif
+};
+
+/* External interrupt init parameters */
+struct no_os_irq_init_param trigger_gpio_irq_params = {
+	.irq_ctrl_id = TRIGGER_GPIO_PIN,
+	.platform_ops = &trigger_gpio_irq_ops,
+	.extra = &trigger_gpio_irq_extra_params
+};
+
+/* External interrupt callback descriptor */
+static struct no_os_callback_desc ext_int_callback_desc = {
+	.callback = burst_capture_callback,
+	.ctx = NULL,
+	.event = NO_OS_EVT_GPIO,
+	.peripheral = NO_OS_GPIO_IRQ
+};
+
+#if (ACTIVE_PLATFORM == STM32_PLATFORM)
+/* PWM GPIO init parameters */
+struct no_os_gpio_init_param pwm_gpio_params = {
+	.port = CNV_PORT_NUM,
+	.number = CNV_PIN_NUM,
+	.platform_ops = &gpio_ops,
+	.extra = &pwm_gpio_extra_init_params
+};
+#endif
+
+/* PWM init parameters for conversion pulses */
+struct no_os_pwm_init_param pwm_init_params = {
+	.id = TIMER1_ID,
+	.period_ns = CONV_TRIGGER_PERIOD_NSEC(SAMPLING_RATE),
+#if (INTERFACE_MODE == SPI_DMA)
+	/* If SPI_DMA is enabled, this pwm is also
+	* also used to trigger a spi tx dma transaction.
+	 * */
```

Comment:
seems ch8  is used to trigger spi tx?

Reply:
It is being used here- https://github.com/analogdevicesinc/precision-converters-firmware/blob/main/projects/ad469x_iio/app/app_config_stm32.c#L217